### PR TITLE
Fix babel error

### PR DESCRIPTION
**Pull request by mengweieric**
### Description

Fix bootstrapping error due to babel error. Changed necessary dependencies versions to compatible ones specific to 2.x. This change is specifically for 2.x, these dependencies are compatible on main.

## Changelog
- fix: Fix babel error

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by sejli — APPROVED** [review]

**Review by virajsanghvi — APPROVED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -174,6 +174,14 @@
...
     chalk "^2.4.2"
 
+"@babel/code-frame@^7.24.7":
+  version "7.24.7"
+  resolved "https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.24.7.tgz#882fd9e09e8ee324e496bd040401c6f046ef4465"
+  integrity sha512-BcYH1CVJBO9tvyIZ2jVeXgSIMvGZ2FDRvDdOIVQyuklNKSsx+eppDEBq/g47Ayw+RqNFE+URvOShmf+f/qwAlA==
+  dependencies:
+    "@babel/highlight" "^7.24.7"
+    picocolors "^1.0.0"
+
```
Comment (AMoo-Miki) [inline yarn.lock:184]: This should have been combined with the block above. Why didn't the resolution do this?

File: yarn.lock [hunk 2]
```diff
@@ -210,13 +218,30 @@
...
+"@babel/generator@^7.25.0":
+  version "7.25.0"
+  resolved "https://registry.yarnpkg.com/@babel/generator/-/generator-7.25.0.tgz#f858ddfa984350bc3d3b7f125073c9af6988f18e"
+  integrity sha512-3LEEcj3PVW8pW2R1SR1M89g/qrYk/m/mB/tLqn7dn4sbBUQyTqnlod+II2U4dqiGtUmkcnAmkMDralTFZttRiw==
+  dependencies:
+    "@babel/types" "^7.25.0"
+    "@jridgewell/gen-mapping" "^0.3.5"
+    "@jridgewell/trace-mapping" "^0.3.25"
+    jsesc "^2.5.1"
+
```
Comment (AMoo-Miki) [inline yarn.lock:230]: Same with this... It should have been combined with the block above. Why is it not?

**Comment by ruanyl** [comment]
Should this backport to 2.x?

File: yarn.lock (hunk 1 above)
Comment (joshuali925) [reply yarn.lock:184]: they shouldn't be combined, 7.22.13 and 7.24.7 are not the same version
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/9431e115011b99ea8b6c87455871a583b464c081/yarn.lock#L169-L170

**Pull request merged**